### ISSUE TITLE
Deprecate "pyarrow-legacy" engine in dask.dataframe.read_parquet

### ISSUE THREAD

**Issue by rjzamora** [dask/dask#8243]
Given the pyarrow version requirements in Dask, we can now assume the `dataset` API will be supported if pyarrow is installed. We can also assume that deprecation warnings will be raised by the pyarrow backend if `dd.read_parquet(..., engine="pyarrow-legacy")` is used.  Therefore, I propose that we add an explicit deprecation warning for the "pyarrow-legacy" engine itself, and establish a timeline for its removal.

Deprecating and removing "pyarrow-legacy" should simplify`read_parquet` maintenance, and should have few (if any) downsides.  However, I welcome pushback if others expect this to cause pain or problems.

**Comment by rjzamora** [comment]
Thanks for the excellent summary @jorisvandenbossche (Sorry for the delay - I know it was a while back). #8758 is taking the first step at addressing this issue (adding a deprecation warning).

**Comment by jorisvandenbossche** [comment]
Small note: there is still some remaining `pq.ParquetDatasetPiece` usage, and that class is also deprecated.

**Comment by jorisvandenbossche** [comment]
No, only the deprecated classes / attributes (which includes ParquetDataset() with `use_legacy_dataset=True` (which gives you the pieces/partitions attributes) and ParquetDatasetPiece). 
But eg `ParquetFile`, `read_table`, `write_table`, `ParquetDataset` (with `use_legacy_dataset=False`, which will become the default), etc are all not going away.

**Comment by jpivarski** [comment]
> cc @jpivarski

I've had to use `pyarrow.parquet.ParquetFile` in a new function ([here](https://github.com/scikit-hep/awkward-1.0/blob/8b107a119caa85936264b48898724ee226ef4d75/src/awkward/_v2/operations/convert/ak_from_parquet.py#L211)) because the Dataset API has no way to specify nested columns (i.e. fields of Structs nested within an object, not the top-level Table fields). See https://github.com/scikit-hep/awkward-1.0/pull/1338#issuecomment-1058782343. In general, `ParquetFile` is a more powerful, though lower-level, reader than the Dataset API.

None of that matters for reading DataFrames, which is what I gather this PR is about. The Dataset API seems to have been designed with DataFrames in mind (the top-level columns are qualitatively different from nested columns).

I think the reason @martindurant was asking is because the function I'm referring to is to be used by [dask-awkward](https://github.com/ContinuumIO/dask-awkward/) to read Parquet in Daskified Awkward Array collections, so it would be new code using it (transitively) in a Dask project.

But I just got your update, @jorisvandenbossche, so it sounds like we have nothing to worry about.
